platform/ios/src/MGLMapView.mm

```diff
@@ -5303,35 +5303,29 @@ - (void)validateLocationServices
 
     if (shouldEnableLocationServices)
     {
-        if (self.locationManager.authorizationStatus == kCLAuthorizationStatusNotDetermined)
-        {
-            BOOL requiresWhenInUseUsageDescription = [NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){11,0,0}];
+        if (self.locationManager.authorizationStatus == kCLAuthorizationStatusNotDetermined) {
             BOOL hasWhenInUseUsageDescription = !![[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationWhenInUseUsageDescription"];
-            BOOL hasAlwaysUsageDescription;
-            if (requiresWhenInUseUsageDescription)
-            {
-                hasAlwaysUsageDescription = !![[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysAndWhenInUseUsageDescription"] && hasWhenInUseUsageDescription;
-            }
-            else
-            {
-                hasAlwaysUsageDescription = !![[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"];
-            }
 
-            if (hasAlwaysUsageDescription)
-            {
-                [self.locationManager requestAlwaysAuthorization];
-            }
-            else if (hasWhenInUseUsageDescription)
-            {
-                [self.locationManager requestWhenInUseAuthorization];
-            }
-            else
-            {
-                NSString *suggestedUsageKeys = requiresWhenInUseUsageDescription ?
-                    @"NSLocationWhenInUseUsageDescription and (optionally) NSLocationAlwaysAndWhenInUseUsageDescription" :
-                    @"NSLocationWhenInUseUsageDescription and/or NSLocationAlwaysUsageDescription";
-                [NSException raise:MGLMissingLocationServicesUsageDescriptionException
-                            format:@"This app must have a value for %@ in its Info.plist.", suggestedUsageKeys];
+            if (@available(iOS 11.0, *)) {
+                // A WhenInUse string is required in iOS 11+ and the map never has any need for Always, so it's enough to just ask for WhenInUse.
+                if (hasWhenInUseUsageDescription) {
```

Comment:
I think we also need to check for ` NSLocationAlwaysAndWhenInUseUsageDescription` here.

Reply:
We do not, as `NSLocationWhenInUseUsageDescription` is required starting in iOS 11 — a developer cannot specify only `NSLocationAlwaysAndWhenInUseUsageDescription`.

When you call `-[CLLocationManager requestAlwaysAuthorization]` or `-[CLLocationManager requestWhenInUseAuthorization]` with only `NSLocationAlwaysAndWhenInUseUsageDescription` in your Info.plist, this is printed to console:

> This app has attempted to access privacy-sensitive data without a usage description. The app's Info.plist must contain both “NSLocationAlwaysAndWhenInUseUsageDescription” and “NSLocationWhenInUseUsageDescription” keys with string values explaining to the user how the app uses this data

The docs for `-[CLLocationManager requestAlwaysAuthorization]` say:

```
 *      Both the NSLocationAlwaysAndWhenInUseUsageDescription and
 *      NSLocationWhenInUseUsageDescription keys must be specified in your
 *      Info.plist; otherwise, this method will do nothing, as your app will be
 *      assumed not to support Always authorization.
```

And `-[CLLocationManager requestWhenInUseAuthorization]` says:

```
 *      The NSLocationWhenInUseUsageDescription key must be specified in your
 *      Info.plist; otherwise, this method will do nothing, as your app will be
 *      assumed not to support WhenInUse authorization.
```

Reply:
Perfect! Thanks for the clarification.

As discussed, let's add tests for these cases in another PR.